crates/ruff_dev/src/generate_options.rs

```diff
@@ -1,9 +1,69 @@
 //! Generate a Markdown-compatible listing of configuration options for `pyproject.toml`.
 //!
 //! Used for <https://docs.astral.sh/ruff/settings/>.
-use itertools::Itertools;
+use std::fmt::Write;
+
 use ruff_workspace::options::Options;
-use ruff_workspace::options_base::{OptionEntry, OptionField};
+use ruff_workspace::options_base::{OptionField, OptionSet, OptionsMetadata, Visit};
+
+pub(crate) fn generate() -> String {
+    let mut output = String::new();
+    generate_set(&mut output, Set::Toplevel(Options::metadata()));
+
+    output
+}
+
+/// Generates a set and returns a bool indicating whether the set had any sub sets
+fn generate_set(output: &mut String, set: Set) {
+    writeln!(output, "### {title}\n", title = set.title()).unwrap();
+
+    let mut visitor = CollectOptionsVisitor::default();
+    set.metadata().record(&mut visitor);
+
+    let (mut fields, mut sets) = (visitor.fields, visitor.groups);
+
+    fields.sort_unstable_by(|(name, _), (name2, _)| name.cmp(name2));
+    sets.sort_unstable_by(|(name, _), (name2, _)| name.cmp(name2));
```

Comment:
I don't know how I feel about sorting options. It disallows grouping options that are semantically related (all resolver settings)